src/main/scala/com/fulcrumgenomics/cmdline/FgBioTool.scala

```diff
@@ -25,10 +25,39 @@ package com.fulcrumgenomics.cmdline
 
 import com.fulcrumgenomics.cmdline.FgBioMain.FailureException
 import com.fulcrumgenomics.sopt.cmdline.ValidationException
+import com.fulcrumgenomics.util.Metric
+import htsjdk.samtools.{SAMFileHeader, SAMProgramRecord}
+
+import scala.annotation.tailrec
+
+/** Stores meta information about the command line use to invoke a tool. */
+case class FgBioToolInfo(name: String, commandLine: String, description: String, version: String) extends Metric {
+  /** Adds a program group to the SAMFileHeader returning the ID. */
+  def applyTo(header: SAMFileHeader): String = {
```

Comment:
It feels a bit weird to have this here in FgBioTool, but I agree it's useful.

Reply:
Also, how about:
1. Calling this `addProgramGroupTo`?
2. Taking either an optional ID or ID prefix?
3. Any reason to return the ID and not the whole record?